filter/filters.py

```diff
@@ -187,6 +197,87 @@ def butter_filter(self, order, cutoffs, btype, name=None):
+        fft_coefs = (1 + c1 * s) / (1 + c2 * s)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('shelf', name, self.sample_freq, coefs)
+    
+    @filter_cache
+    def analog_lp1(self, center_freq, name=None):
+        s_norm = 1j * self.freq_scale / center_freq
+        fft_coefs = 1. / (1 + s_norm)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('lp1', name, self.sample_freq, coefs)
+    
+    @filter_cache
+    def analog_lp2(self, center_freq, Q, name=None):
+        s_norm = 1j * self.freq_scale / center_freq
+        fft_coefs = 1. / (1 + s_norm / Q + s_norm ** 2)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('lp2', name, self.sample_freq, coefs)
+        
+    @filter_cache
+    def analog_hp1(self, center_freq, name=None):
+        s_norm = 1j * self.freq_scale / center_freq
+        fft_coefs = s_norm / (1 + s_norm)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('hp1', name, self.sample_freq, coefs)
+        
+    @filter_cache
+    def analog_hp2(self, center_freq, Q, name=None):
+        s_norm = 1j * self.freq_scale / center_freq
+        fft_coefs = (s_norm ** 2) / (1 + s_norm / Q + s_norm ** 2)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('hp2', name, self.sample_freq, coefs)
+    
+    @filter_cache
+    def delay_sec(self, time, name=None):
+        s = 1j * 2 * numpy.pi * self.freq_scale
+        fft_coefs = numpy.exp(-s * time)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('delay', name, self.sample_freq, coefs)
+    
+    @filter_cache
+    def delay_deg(self, freq_hz, phase_deg, name=None):
+        return self.delay_sec(phase_deg / 360. / freq_hz)
+
+    @filter_cache
+    def spectral_slope(self, start_freq, stop_freq, slope_db_dec, name=None):
+        assert start_freq > 0
+        assert stop_freq > start_freq
+        fa = numpy.abs(self.freq_scale)
+        width_dec = numpy.log10(stop_freq / start_freq)
+        lf_gain_db = -0.5 * width_dec * slope_db_dec
+        hf_gain_db = 0.5 * width_dec * slope_db_dec
+        filt_fft_mag = numpy.zeros(self.filter_size)
+        filt_fft_mag[fa <= start_freq] = 10. ** (lf_gain_db / 20.)
+        filt_fft_mag[fa >= stop_freq] = 10. ** (hf_gain_db / 20.)
+        active_mask = (fa > start_freq) * (fa < stop_freq)
+        filt_fft_mag[active_mask] = 10. ** ((lf_gain_db + slope_db_dec * numpy.log10(fa[active_mask] / start_freq)) / 20.)
+        fft_coefs = self.hilbert_fft_coefs_from_mag(filt_fft_mag)
+        coefs = numpy.real(scipy.fftpack.ifft(fft_coefs))
+        return Filter('slope', name, self.sample_freq, coefs)
+
+    @filter_cache
+    def parametric_eq(self, center_freq, Q, gain_db, name=None):
+        s = 1j * self.freq_scale / center_freq
+        B = 1. / Q
+        g = 10. ** (gain_db / 20.)
+        H = (s ** 2 + g * B * s + 1) / (s ** 2 + B * s + 1)
```

Comment:
Please see https://ccrma.stanford.edu/~jos/filters/Peaking_Equalizers.html.  It's a 2nd order analog filter that can be implemented with an RLC network (the inductor is usually synthesized by a gyrator in an active circuit).  I was expecting that you'd use the FIR filters to flatten out driver responses and that this would be more useful for voicing (i.e. playing around with intentional frequency response contours to improve subjective performance).  But there's no harm in mixing them up however you want.